design_system/src/commonMain/kotlin/net/thechance/mena/designsystem/presentation/component/appBar/AppBar.kt

```diff
@@ -0,0 +1,143 @@
+package net.thechance.mena.designsystem.presentation.component.appBar
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import mena.design_system.generated.resources.Res
+import mena.design_system.generated.resources.checkmark
+import mena.design_system.generated.resources.ic_arrow_left
+import mena.design_system.generated.resources.ic_user
+import net.thechance.mena.designsystem.presentation.theme.theme.MenaTheme
+import net.thechance.mena.designsystem.presentation.theme.theme.Theme
+import org.jetbrains.compose.resources.painterResource
+import org.jetbrains.compose.ui.tooling.preview.Preview
+
+@Composable
+fun AppBar(
+    title: String,
+    modifier: Modifier = Modifier,
+    titleColor: Color = Theme.colorScheme.shadePrimary,
+    contentPadding: PaddingValues = PaddingValues(horizontal = 16.dp, vertical = 18.dp),
+    contentTitlePadding: PaddingValues = PaddingValues(horizontal = 8.dp, vertical = 6.dp),
+    leadingContent: (@Composable () -> Unit)? = null,
```

Comment:
I think the Back Button should be part of the component, not passed from outside, with the option to hide it
```suggestion
    withBackButton: Boolean = ture,
```

Reply:
Thanks for reviewing this part but don't you think this will minimize the reusability of this component??

Reply:
What if the leading Icon or content changes to another content ?